File: src/main/java/dev/fdp/races/VisualsManager.java
package dev.fdp.races;

import dev.fdp.races.visualUpdaters.EarsUpdater;
import dev.fdp.races.visualUpdaters.IVisualUpdater;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

import java.util.List;

public class VisualsManager {
    private static final List<IVisualUpdater> visualUpdaters = List.of(
            new EarsUpdater()
    );

    public static void updateVisualsForPlayer(Player player) {
        RaceManager raceManager = FDP_Races.getInstance().raceManager;
        Race race = raceManager.getRaceForPlayer(player);

        for (IVisualUpdater updater : visualUpdaters) {
            updater.updateVisuals(player, race);
        }
    }

    public static void reloadVisualsForAllPlayer() {
        for (Player player : Bukkit.getOnlinePlayers()) {
            updateVisualsForPlayer(player);
        }
    }

    public static void unloadVisualsForPlayer(Player player) {
        for (IVisualUpdater updater : visualUpdaters) {
            updater.unloadVisuals(player);
        }
    }
}

File: src/main/java/dev/elysium/eraces/VisualsManager.kt
package dev.elysium.eraces

import dev.elysium.eraces.ERaces.Companion.getInstance
import dev.elysium.eraces.visualUpdaters.EarsUpdater
import dev.elysium.eraces.visualUpdaters.IVisualUpdater
import org.bukkit.Bukkit
import org.bukkit.entity.Player

object VisualsManager {
    private val visualUpdaters = listOf<IVisualUpdater>(
        EarsUpdater(),
    ) as MutableList<IVisualUpdater>

    @JvmStatic
    fun updateVisualsForPlayer(player: Player) {
        val race = getInstance().context.playerDataManager.getPlayerRace(player)

        for (updater in visualUpdaters) updater.updateVisuals(race, player)
    }

    fun reloadVisualsForAllPlayer() {
        for (player in Bukkit.getOnlinePlayers()) updateVisualsForPlayer(player)
    }

    fun unloadVisualsForPlayer(player: Player) {
        for (updater in visualUpdaters) updater.unloadVisuals(player)
    }
}